src/native/eventpipe/ep-buffer-manager.c

```diff
@@ -897,6 +897,9 @@ ep_buffer_manager_write_event (
 	// Before we pick a buffer, make sure the event is enabled.
 	ep_return_false_if_nok (ep_event_is_enabled (ep_event));
 
+	// Check that the payload size is less than 64 KB (max size for ETW events)
+	ep_return_false_if_nok (ep_event_payload_get_size (payload) <= 64 * 1024);
```

Comment:
ETW might limit its payloads to 64KB, but does EventPipe need to? `EventSource` doesn't explicitly prevent you from having payloads this large, e.g., your test.

Reply:
It doesn't. But ETW will drop the event if it's greater than 64KB and I'd like to enforce the same behavior here since:
1. We need some kind of limit regardless. Currently >100KB payloads crash the app. We could enforce a 100KB limit instead of 64KB.
2. But if we do enforce 100KB limit, that leads to a discrepancy between ETW/EventPipe behavior which we then need to go and figure out how customers can find out whether big events are getting dropped and document why such behavior differences exist.
3. Another approach is to enforce 1MB payload (which we currently do not explicitly but because we just fail to allocate buffers greater than that) and increase the max flushing block size to 1MB but that's a pretty inefficient use of space for sessions listening on much smaller-sized payloads (i.e. counters).

Reply:
Fair point. Let me mull this over, but I think 64KB is fine. Should we remove the `DebugBreak` in the `EP_UNLIKELY` macro expansion?

Reply:
I'd actually like to keep that. It lets us catch exceptions like this rather than silently failing and us never noticing the failures.